pkgs/tools/networking/curl-impersonate/default.nix

```diff
@@ -1,27 +1,190 @@
+
+    preConfigure = ''
+      export GOCACHE=$TMPDIR/go-cache
+      export GOPATH="$TMPDIR/go"
+      export GOPROXY=file://${passthru.boringssl-go-modules}
+      export GOSUMDB=off
+    '';
+
+    postInstall = ''
+      # Patch all shebangs of installed scripts
+      patchShebangs $out/bin
+
+      # Build and install completions for each curl binary
+      for bin in $out/bin/curl-impersonate-*; do
+        if [[ "$bin" =~ '-config' ]]; then
+          continue
+        fi
+
+        binname="$(basename "$bin")"
+
+        # Patch in correct binary name and alias it to all scripts
+        perl curl-*/scripts/completion.pl --curl $bin --shell zsh >"$TMPDIR/$binname".zsh
+        substituteInPlace "$TMPDIR/$binname".zsh \
+          --replace \
+            '#compdef curl' \
+            "#compdef $binname$(find $out/bin -name 'curl_*' -printf " %f=$binname")"
+
+        perl curl-*/scripts/completion.pl --curl $bin --shell fish >"$TMPDIR/$binname".fish
+        substituteInPlace "$TMPDIR/$binname".fish \
+          --replace \
+            '--command curl' \
+            "--command $binname$(find $out/bin -name 'curl_*' -printf " --command %f")"
+
+        # Install zsh and fish completions
+        installShellCompletion "$TMPDIR/$binname".{zsh,fish}
+      done
+    '';
+
+    preFixup = ''
+      # If libnssckbi.so is needed, link libnssckbi.so without needing nss in closure
+      if grep -F nssckbi $out/lib/libcurl-impersonate-*${stdenv.hostPlatform.extensions.sharedLibrary}; then
+        ln -s ${p11-kit}/lib/pkcs11/p11-kit-trust${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libnssckbi${stdenv.hostPlatform.extensions.sharedLibrary}
+        ${lib.optionalString (stdenv.hostPlatform.parsed.kernel.execFormat == lib.systems.parse.execFormats.elf)
+          "patchelf --add-needed libnssckbi${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/libcurl-impersonate-*${stdenv.hostPlatform.extensions.sharedLibrary}"}
+      fi
+    '';
+
+    disallowedReferences = [ go ];
+
+    passthru = {
+      deps = callPackage ./deps.nix {};
+
+      boringssl-go-modules = (buildGoModule {
+        inherit (passthru.deps."boringssl.zip") name;
+
+        src = passthru.deps."boringssl.zip";
+        vendorHash = "sha256-ISmRdumckvSu7hBXrjvs5ZApShDiGLdD3T5B0fJ1x2Q=";
+
+        nativeBuildInputs = [ unzip ];
+
+        proxyVendor = true;
+      }).go-modules;
+    };
+
+    meta = with lib; {
+      description = "A special build of curl that can impersonate Chrome & Firefox";
+      homepage = "https://github.com/lwthiker/curl-impersonate";
+      license = with licenses; [ curl mit ];
+      maintainers = with maintainers; [ deliciouslytyped lilyinstarlight ];
+      platforms = platforms.unix;
+      # Seems to need newer SDK for x86_64-darwin and libtool complains about missing "-static" flag on both x86_64-darwin and aarch64-darwin
```

Comment:
Did you try to use the apple 11 sdk callPackage wrapper?

Reply:
I would, but I really don't have a Mac personally and don't have the time to debug the libtool issue either. I mostly left the note there in case someone else who wants the package on Darwin can tackle the issue

Reply:
It turns out the darwin build was easy to fix: the `libtool` bit turned out to be needing the `libtool` in `darwin.cctools` (so it ended up needing to shadow `libtool` from autotools) and otherwise the only other thing needed was the sdk11 `callPackage` like you mentioned

I tested the build in GitHub Actions on x86_64-darwin but didn't actually test that the binaries fully worked